app/web/features/auth/notifications/PushNotificationPermission.tsx

```diff
@@ -0,0 +1,161 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import CustomColorSwitch from "components/CustomColorSwitch";
+import { Trans, useTranslation } from "i18n";
+import { AUTH } from "i18n/namespaces";
+import { useState } from "react";
+import {
+  getVapidPublicKey,
+  registerPushNotificationSubscription,
+} from "service/notifications";
+import { theme } from "theme";
+import { arrayBufferToBase64 } from "utils/arrayBufferToBase64";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  alert: {
+    marginBottom: theme.spacing(3),
+    marginTop: theme.spacing(2),
+  },
+  status: {
+    marginBottom: theme.spacing(2),
+  },
+  titleBox: {
+    display: "flex",
+    alignItems: "center",
+  },
+}));
+
+export default function PushNotificationPermission({
+  className,
+}: {
+  className: string;
+}) {
+  const { t } = useTranslation(AUTH);
+  const classes = useStyles();
+
+  const [permission, setPermission] = useState(Notification.permission);
```

Comment:
So the value of Notification.permission is stored in the browser's internal state. But you're right, if I deny it right now the switch won't enable it again. 

This is because the `Notification.requestPermission()` method only works when the permission status is "default" (i.e., when the user hasn't already made a choice). If the permission status is "denied", the browser will not ask for permission again automatically (unfortunately for us). Once the user denies the permission, it is stored at the browser level, and the browser does not allow the web app to prompt the user again.


As a better solution, I just added a check that will determine what browser and OS the user is using and show an error message specifically with instructions for how to enable notifications again for that browser.

Reply:
Ugh okay after reading about this more I realized unsubscribing the user from push notifications with `.unsubscribe()` but it won't update `Notification.permission`.

Basically granting permission is a one way street and disabling it or reactivating again has to be done via the browser.

So I think the way to go is just have the switch to enable them, and give instructions for how to disable or reenable in the browser for the other scenarios. Annoying.

Reply:
I mean once we have permission, then we don't need to worry about that anymore: we'll just subscribe/unsubscribe the user, right?

Reply:
Okay gonna resolve this since you are right and we talked about it on a zoom call. I'm gonna refactor this to base itself on the subscription being null or not rather than the permission.